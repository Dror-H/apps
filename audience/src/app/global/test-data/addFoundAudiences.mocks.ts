import { SearchResult, SupportedProviders } from '@instigo-app/data-transfer-object';

export const addFoundAudiencesSearchResults1: SearchResult[] = [
  {
    id: '14b4016c-41dc-47cb-85e3-c61ee3650fdc',
    spec: {
      provider: SupportedProviders.FACEBOOK,
      include: {
        and: [
          { or: { ageRange: [{ providerId: '(30,60)', type: 'ageRange' }] } },
          { or: { locales: [{ providerId: [] }] } },
          { or: {} },
        ],
      },
      exclude: { or: {} },
    },
    topics: [],
    rank: 100,
    specRatio: { interests: 100, demographics: 0, behaviors: 0 },
    userTags: [],
  },
  {
    id: '24b4016c-41dc-47cb-85e3-c61ee3650fdg',
    spec: {
      provider: SupportedProviders.FACEBOOK,
      include: {
        and: [
          { or: { ageRange: [{ providerId: '(30,60)', type: 'ageRange' }] } },
          { or: { locales: [{ providerId: [] }] } },
          { or: {} },
        ],
      },
      exclude: { or: {} },
    },
    topics: [],
    rank: 50,
    specRatio: { interests: 100, demographics: 0, behaviors: 0 },
    userTags: [],
  },
];
export const addFoundAudiencesSearchResults2: SearchResult[] = [
  {
    id: '34b4016c-41dc-47cb-85e3-c61ee3651234',
    spec: {
      provider: SupportedProviders.FACEBOOK,
      include: {
        and: [
          { or: { ageRange: [{ providerId: '(30,60)', type: 'ageRange' }] } },
          { or: { locales: [{ providerId: [] }] } },
          { or: {} },
        ],
      },
      exclude: { or: {} },
    },
    topics: [],
    rank: 75,
    specRatio: { interests: 100, demographics: 0, behaviors: 0 },
    userTags: [],
  },
];

export const addFoundAudiencesSortedResult: SearchResult[] = [
  addFoundAudiencesSearchResults1[0],
  addFoundAudiencesSearchResults2[0],
  addFoundAudiencesSearchResults1[1],
];
