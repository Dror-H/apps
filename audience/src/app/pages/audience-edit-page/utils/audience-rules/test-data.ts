import { AudienceDto, SupportedProviders, TargetingDto } from '@instigo-app/data-transfer-object';

export const mockRegularTargeting: TargetingDto = {
  provider: SupportedProviders.FACEBOOK,
  include: {
    and: [
      { or: { ageRange: [{ providerId: '(25,50)', type: 'ageRange' }] } },
      { or: { locales: [{ providerId: [] }] } },
      {
        or: {
          behaviors: [
            {
              providerId: '6002714898572',
              type: 'behaviors',
              path: ['Behaviours', 'Digital activities', 'Small business owners'],
              name: 'Small business owners',
              audienceSize: 48970322,
            },
          ],
          interests: [
            {
              providerId: '6002884511422',
              type: 'interests',
              path: ['Interests', 'Business and industry', 'Small business'],
              name: 'Small business',
              audienceSize: 192686100.5,
            },
          ],
          employers: [
            {
              providerId: '109288035760640',
              type: 'employers',
              path: ['Demographics', 'Work', 'Employers', 'Business Owner'],
              name: 'Business Owner',
              audienceSize: 116699,
            },
          ],
        },
      },
      {
        or: {
          interests: [
            {
              providerId: '6002964301046',
              type: 'interests',
              path: ['Interests', 'Additional interests', 'Business school'],
              name: 'Business school',
              audienceSize: 41083805,
            },
          ],
          fieldsOfStudy: [
            {
              providerId: '106335129403847',
              type: 'fieldsOfStudy',
              path: ['Demographics', 'Education', 'Fields of study', 'Business management'],
              name: 'Business management',
              audienceSize: 2636510.5,
            },
          ],
          educationStatuses: [{ type: 'educationStatuses', providerId: '' }],
        },
      },
      {
        or: {
          interests: [
            {
              providerId: '6002964301046',
              type: 'interests',
              path: ['Interests', 'Additional interests', 'Business school'],
              name: 'Business school',
              audienceSize: 41083805,
            },
          ],
          employers: [
            {
              providerId: '116050848438791',
              type: 'employers',
              path: ['Demographics', 'Work', 'Employers', 'Currently looking for a job.'],
              name: 'Currently looking for a job.',
              audienceSize: 3339.5,
            },
          ],
        },
      },
    ],
  },
  exclude: {
    or: {
      behaviors: [
        {
          providerId: '6071631541183',
          type: 'behaviors',
          path: ['Behaviours', 'Purchase behaviour', 'Engaged shoppers'],
          name: 'Engaged shoppers',
          audienceSize: 539317289.5,
        },
        {
          providerId: '123456',
          type: 'behaviors',
          path: ['Behaviours'],
          name: 'test',
          audienceSize: 123456,
        },
      ],
      demographics: [
        {
          providerId: '654321',
          type: 'demographics',
          path: ['Demographics'],
          name: 'test 2',
          audienceSize: 654321,
        },
      ],
      interests: [
        {
          providerId: '123654',
          type: 'interests',
          path: ['Interests'],
          name: 'test 2',
          audienceSize: 321654,
        },
      ],
    },
  },
};

export const mockRegularTargetingWithLocation: TargetingDto = {
  provider: SupportedProviders.FACEBOOK,
  include: {
    and: [
      { or: { ageRange: [{ providerId: '(25,50)', type: 'ageRange' }] } },
      { or: { locales: [{ providerId: [] }] } },
      { or: { locationTypes: [], countries: [] } },
      {
        or: {
          behaviors: [
            {
              providerId: '6002714898572',
              type: 'behaviors',
              path: ['Behaviours', 'Digital activities', 'Small business owners'],
              name: 'Small business owners',
              audienceSize: 48970322,
            },
          ],
          interests: [
            {
              providerId: '6002884511422',
              type: 'interests',
              path: ['Interests', 'Business and industry', 'Small business'],
              name: 'Small business',
              audienceSize: 192686100.5,
            },
          ],
          employers: [
            {
              providerId: '109288035760640',
              type: 'employers',
              path: ['Demographics', 'Work', 'Employers', 'Business Owner'],
              name: 'Business Owner',
              audienceSize: 116699,
            },
          ],
        },
      },
      {
        or: {
          interests: [
            {
              providerId: '6002964301046',
              type: 'interests',
              path: ['Interests', 'Additional interests', 'Business school'],
              name: 'Business school',
              audienceSize: 41083805,
            },
          ],
          fieldsOfStudy: [
            {
              providerId: '106335129403847',
              type: 'fieldsOfStudy',
              path: ['Demographics', 'Education', 'Fields of study', 'Business management'],
              name: 'Business management',
              audienceSize: 2636510.5,
            },
          ],
          educationStatuses: [{ type: 'educationStatuses', providerId: '' }],
        },
      },
      {
        or: {
          interests: [
            {
              providerId: '6002964301046',
              type: 'interests',
              path: ['Interests', 'Additional interests', 'Business school'],
              name: 'Business school',
              audienceSize: 41083805,
            },
          ],
          employers: [
            {
              providerId: '116050848438791',
              type: 'employers',
              path: ['Demographics', 'Work', 'Employers', 'Currently looking for a job.'],
              name: 'Currently looking for a job.',
              audienceSize: 3339.5,
            },
          ],
        },
      },
    ],
  },
  exclude: {
    or: {
      behaviors: [
        {
          providerId: '6071631541183',
          type: 'behaviors',
          path: ['Behaviours', 'Purchase behaviour', 'Engaged shoppers'],
          name: 'Engaged shoppers',
          audienceSize: 539317289.5,
        },
        {
          providerId: '123456',
          type: 'behaviors',
          path: ['Behaviours'],
          name: 'test',
          audienceSize: 123456,
        },
      ],
      demographics: [
        {
          providerId: '654321',
          type: 'demographics',
          path: ['Demographics'],
          name: 'test 2',
          audienceSize: 654321,
        },
      ],
      interests: [
        {
          providerId: '123654',
          type: 'interests',
          path: ['Interests'],
          name: 'test 2',
          audienceSize: 321654,
        },
      ],
    },
  },
};
