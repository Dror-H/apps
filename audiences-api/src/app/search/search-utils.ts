import { flatten, uniqBy } from 'lodash';
import { FlexibleSpec, SearchResult, Segment } from '../shared/model';

export class SearchUtils {
  static removeEmptyObjectsAndArraysFromTargeting(audiences: SearchResult[]): void {
    for (const audience of audiences) {
      this.removeEmptyObjectsAndArraysFromFlexibleSpecs(audience);
      this.removeEmptyObjectsAndArraysFromExclusions(audience);
    }
  }

  static removeEmptyObjectsAndArraysFromFlexibleSpecs(audience: SearchResult): void {
    const flexSpecs = audience.spec.targeting.flexible_spec;
    for (const flexSpec of flexSpecs) {
      for (const key in flexSpec) {
        if (flexSpec[key].length === 0) {
          delete flexSpec[key];
        }
      }
    }
    audience.spec.targeting.flexible_spec = flexSpecs.filter((flexSpec) => Object.keys(flexSpec).length > 0);
  }

  static removeEmptyObjectsAndArraysFromExclusions(audience: SearchResult): void {
    let exclusions = audience.spec.targeting.exclusions;
    if (exclusions) {
      for (const key in exclusions) {
        if (exclusions[key].length === 0) {
          delete exclusions[key];
        }
      }
      if (Object.keys(exclusions).length === 0) {
        exclusions = undefined;
      }
    }
    audience.spec.targeting.exclusions = exclusions;
  }

  static removeInvalidTypeSegments(audiences: SearchResult[]): void {
    const invalidSpecKeys: (keyof FlexibleSpec)[] = ['college_years'];
    for (const audience of audiences) {
      this.removeInvalidTypesFlexSpec(audience, invalidSpecKeys);
      this.removeInvalidTypesExclusion(audience, invalidSpecKeys);
    }
  }

  static removeInvalidTypesFlexSpec(audience: SearchResult, invalidSpecKeys: (keyof FlexibleSpec)[]): void {
    const flexSpecs = audience.spec.targeting.flexible_spec;
    const validFlexSpecs = flexSpecs.map((flexSpec) => {
      for (const key in flexSpec) {
        if (invalidSpecKeys.includes(key as keyof FlexibleSpec)) {
          delete flexSpec[key];
        }
      }
      return flexSpec;
    });
    audience.spec.targeting.flexible_spec = validFlexSpecs;
  }

  static removeInvalidTypesExclusion(audience: SearchResult, invalidSpecKeys: (keyof FlexibleSpec)[]): void {
    const exclusions = audience.spec.targeting.exclusions;
    if (exclusions) {
      for (const key in exclusions) {
        if (invalidSpecKeys.includes(key as keyof FlexibleSpec)) {
          delete audience.spec.targeting.exclusions[key];
        }
      }
    }
  }

  static getAudiencesSegments(audiences: SearchResult[], returnExclusions = false): Segment[] {
    let segments: Segment[] = [];
    for (const audience of audiences) {
      segments = segments.concat(this.getAudienceFlexSpecSegments(audience.spec.targeting.flexible_spec));
      if (returnExclusions) {
        segments = segments.concat(this.getAudienceExclusionSegments(audience.spec.targeting.exclusions));
      }
    }
    return uniqBy(segments, 'id');
  }

  static getAudienceFlexSpecSegments(flexSpecs: FlexibleSpec[]): Segment[] {
    const segments: Segment[] = [];
    if (flexSpecs.length > 0) {
      for (const flexSpec of flexSpecs) {
        const flexSpecSegments = flatten(Object.values(flexSpec)) as Segment[];
        for (const segment of flexSpecSegments) {
          segments.push(segment);
        }
      }
    }
    return segments;
  }

  static getAudienceExclusionSegments(exclusions?: FlexibleSpec): Segment[] {
    const segments: Segment[] = [];
    if (exclusions && Object.keys(exclusions).length > 0) {
      const exclusionSegments = flatten(Object.values(exclusions)) as Segment[];
      for (const segment of exclusionSegments) {
        segments.push(segment);
      }
    }
    return segments;
  }

  static getValidSegmentsForValidationCall(segments: Segment[]): Segment[] {
    const invalidSegmentTypesForValidationCall = ['demographics'];
    return segments.filter((segment) => !invalidSegmentTypesForValidationCall.includes(segment.type));
  }

  static filterInvalidSegmentsFromFb(audiences: SearchResult[], validSegmentIds: string[]): void {
    const invalidSegmentTypesForValidationCall = ['demographics'];
    for (const audience of audiences) {
      this.removeInvalidFlexibleSpecFromFb(audience, validSegmentIds, invalidSegmentTypesForValidationCall);
      this.removeInvalidExclusionsFromFb(audience, validSegmentIds, invalidSegmentTypesForValidationCall);
    }
  }

  static removeInvalidFlexibleSpecFromFb(
    audience: SearchResult,
    validSegmentIds: string[],
    invalidSegmentTypesForValidationCall: string[],
  ): void {
    const flexSpec = audience.spec.targeting.flexible_spec;
    for (let i = 0; i < flexSpec.length; i++) {
      const spec = flexSpec[i];
      for (const key in spec) {
        if (!invalidSegmentTypesForValidationCall.includes(key)) {
          audience.spec.targeting.flexible_spec[i][key] = audience.spec.targeting.flexible_spec[i][key].filter((f) =>
            validSegmentIds.includes(f.id),
          );
        }
      }
    }
  }

  static removeInvalidExclusionsFromFb(
    audience: SearchResult,
    validSegmentIds: string[],
    invalidSegmentTypesForValidationCall: string[],
  ): void {
    const exclusions = audience.spec.targeting.exclusions;
    for (const key in exclusions) {
      if (!invalidSegmentTypesForValidationCall.includes(key)) {
        audience.spec.targeting.exclusions[key] = audience.spec.targeting.exclusions[key].filter((f) =>
          validSegmentIds.includes(f.id),
        );
      }
    }
  }
}
