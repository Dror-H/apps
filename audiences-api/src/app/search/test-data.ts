export const mockSearchResults = [
  {
    id: '4870eab0-7ed2-56b1-9493-5b4684a7b03c',
    createdByContributors: false,
    spec: {
      id: '0478f7d5-55c7-4106-ba34-4f691236969c',
      name: '',
      targeting: {
        age_max: 65,
        age_min: 18,
        genders: [0],
        locales: [],
        exclusions: {},
        flexible_spec: [
          {
            work_positions: [
              {
                id: '105703889464434',
                name: 'Motivational speaker',
              },
              {
                id: '839841992748313',
                name: 'Personal Coach',
              },
            ],
          },
        ],
        geo_locations: {
          countries: ['US'],
          location_types: ['home', 'recent'],
        },
        excluded_geo_locations: {},
        targeting_optimization: 'none',
        excluded_custom_audiences: [],
      },
    },
    segments: [],
    userTags: [],
    topics: [],
  },
  {
    id: '478605de-ba43-5d8c-9c52-b1043e4cc0b7',
    createdByContributors: false,
    spec: {
      id: '23848984859650054',
      name: 'Interest Jewellry - UK - 18+ – Copy',
      campaign: {
        id: '23848984859400054',
        name: 'SO - Conversions - Bracelet - CBO - UK Only',
      },
      targeting: {
        age_max: 65,
        age_min: 35,
        genders: [2],
        exclusions: {
          interests: [
            {
              id: '6003109292833',
              name: 'Drop shipping',
            },
            {
              id: '6019154143076',
              name: 'AliExpress',
            },
          ],
        },
        flexible_spec: [
          {
            interests: [
              {
                id: '6003266225248',
                name: 'Jewellery',
              },
            ],
          },
        ],
        geo_locations: {
          countries: ['GB'],
          location_types: ['home', 'recent'],
        },
        brand_safety_content_filter_levels: ['FACEBOOK_STANDARD', 'AN_STANDARD'],
      },
    },
    segments: ['SO - Conversions - Bracelet - CBO - UK Only', 'Interest Jewellry - UK - 18+ – Copy', 'Jewellery'],
    userTags: [],
    topics: ['uk', 'jewellry', 'jewellery', 'interest', 'copy'],
  },
];

export const mockEmptyFlexSpecAndExclusions = {
  id: '478605de-ba43-5d8c-9c52-b1043e4cc0b7',
  createdByContributors: false,
  spec: {
    id: '23848984859650054',
    name: 'Interest Jewellry - UK - 18+ – Copy',
    campaign: {
      id: '23848984859400054',
      name: 'SO - Conversions - Bracelet - CBO - UK Only',
    },
    targeting: {
      age_max: 65,
      age_min: 35,
      genders: [2],
      exclusions: {
        interests: [],
        work_employers: [],
      },
      flexible_spec: [
        {
          interests: [],
          work_positions: [],
        },
      ],
      geo_locations: {
        countries: ['GB'],
        location_types: ['home', 'recent'],
      },
      brand_safety_content_filter_levels: ['FACEBOOK_STANDARD', 'AN_STANDARD'],
    },
  },
  segments: ['SO - Conversions - Bracelet - CBO - UK Only', 'Interest Jewellry - UK - 18+ – Copy', 'Jewellery'],
  userTags: [],
  topics: ['uk', 'jewellry', 'jewellery', 'interest', 'copy'],
};
