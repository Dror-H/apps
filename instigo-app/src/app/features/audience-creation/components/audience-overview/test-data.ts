import { ProviderList } from './provider';

export const providerListSelectionConfig: ProviderList = {
  config: { selectedColumns: ['email', 'phone', 'remove me column'] },
  tableConfiguration: {
    scrollbarH: true,
    tableId: `facebook_audience_table`,
    columns: [
      { name: 'email', prop: 'email' },
      { name: 'phone', prop: 'phone' },
      { name: 'remove me column', prop: 'remove me column' },
    ],
    selectable: false,
    columnsCustomizable: false,
    clientSide: true,
    searchable: false,
    matches: () => true,
  },
  tableState: {},
  tableData: [],
};

export const providerListMapColumnConfig: ProviderList = {
  config: { selectedColumns: ['email', 'phone', 'map me'] },
  tableConfiguration: {
    scrollbarH: true,
    tableId: `facebook_audience_table`,
    columns: [
      { name: 'email', prop: 'email' },
      { name: 'phone', prop: 'phone' },
      { name: 'map me', prop: 'map me' },
    ],
    selectable: false,
    columnsCustomizable: false,
    clientSide: true,
    searchable: false,
    matches: () => true,
  },
  tableState: {},
  tableData: [],
};
