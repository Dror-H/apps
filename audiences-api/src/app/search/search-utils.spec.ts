import { cloneDeep } from 'lodash';
import { SearchUtils } from './search-utils';
import { mockEmptyFlexSpecAndExclusions, mockSearchResults } from './test-data';

describe('search-utils', () => {
  afterEach(() => {
    jest.restoreAllMocks();
  });

  describe('removeEmptyObjectsAndArraysFromTargeting', () => {
    it('should call removeEmptyObjectsAndArraysFromFlexibleSpecs and removeEmptyObjectsAndArraysFromExclusions for each audience with each audience', () => {
      const removeEmptyFlexSpecsSpy = jest
        .spyOn(SearchUtils, 'removeEmptyObjectsAndArraysFromFlexibleSpecs')
        .mockImplementation();
      const removeEmptyExclusionsSpy = jest
        .spyOn(SearchUtils, 'removeEmptyObjectsAndArraysFromExclusions')
        .mockImplementation();
      const mockAudiences = ['test 1', 'test 2', 'test 3'];
      SearchUtils.removeEmptyObjectsAndArraysFromTargeting(mockAudiences as any);
      mockAudiences.forEach((mockAud, i) => {
        expect(removeEmptyFlexSpecsSpy).toBeCalledWith(mockAudiences[i]);
        expect(removeEmptyExclusionsSpy).toBeCalledWith(mockAudiences[i]);
      });
    });

    it('should remove empty objects and arrays from flexible specs and exclusions', () => {
      SearchUtils.removeEmptyObjectsAndArraysFromTargeting([mockEmptyFlexSpecAndExclusions] as any);
      expect(mockEmptyFlexSpecAndExclusions).toMatchSnapshot();
    });
  });

  describe('removeInvalidTypeSegments', () => {
    it('should call removeInvalidTypesFlexSpec and removeInvalidTypesExclusion with each audience', () => {
      const audiencesMock = ['mock audience 1', 'mock audience 2'];
      const removeInvalidTypesFlexSpecSpy = jest.spyOn(SearchUtils, 'removeInvalidTypesFlexSpec').mockImplementation();
      const removeInvalidTypesExclusionSpy = jest
        .spyOn(SearchUtils, 'removeInvalidTypesExclusion')
        .mockImplementation();
      (SearchUtils as any).removeInvalidTypeSegments(audiencesMock);
      audiencesMock.forEach((audienceMock, i) => {
        expect(removeInvalidTypesFlexSpecSpy.mock.calls[i][0]).toBe(audienceMock);
        expect(removeInvalidTypesExclusionSpy.mock.calls[i][0]).toBe(audienceMock);
      });
    });
  });
  describe('removeInvalidTypesFlexSpec', () => {
    it('should remove audience.spec.targeting.flexible_spec at keys from argument', () => {
      const mockAudience = { spec: { targeting: { flexible_spec: [{ testKey: 'test' }] } } };
      (SearchUtils as any).removeInvalidTypesFlexSpec(mockAudience, ['testKey']);
      expect(mockAudience.spec.targeting.flexible_spec[0].testKey).toBeUndefined();
    });
  });
  describe('removeInvalidTypesExclusion', () => {
    it('should remove audience.spec.targeting.exclusions at keys from argument', () => {
      const mockAudience = { spec: { targeting: { exclusions: { testKey: 'test' } } } };
      (SearchUtils as any).removeInvalidTypesExclusion(mockAudience, ['testKey']);
      expect(mockAudience.spec.targeting.exclusions.testKey).toBeUndefined();
    });
  });

  describe('filterInvalidSegmentsFromFb', () => {
    it('should call removeInvalidFlexibleSpecFromFb and removeInvalidExclusionsFromFb for each with same arguments as given', () => {
      const args = [
        ['test 1', 'test 2'],
        ['validID 1', 'validID 2'],
      ];
      const removeInvalidFlexibleSpecFromFbSpy = jest
        .spyOn(SearchUtils, 'removeInvalidFlexibleSpecFromFb')
        .mockImplementation();
      const removeInvalidExclusionsFromFb = jest
        .spyOn(SearchUtils, 'removeInvalidExclusionsFromFb')
        .mockImplementation();
      (SearchUtils as any).filterInvalidSegmentsFromFb(...args);
      args[0].forEach((mockAud, i) => {
        expect(removeInvalidFlexibleSpecFromFbSpy).toBeCalledWith(args[0][i], args[1], ['demographics']);
        expect(removeInvalidExclusionsFromFb).toBeCalledWith(args[0][i], args[1], ['demographics']);
      });
    });
  });
  describe('removeInvalidFlexibleSpecFromFb', () => {
    it('should remove invalid segments from flexible spec', () => {
      const mockAudience = cloneDeep(mockSearchResults[0]);
      SearchUtils.removeInvalidFlexibleSpecFromFb(mockAudience as any, ['105703889464434'], ['demographics']);
      expect(mockAudience).toMatchSnapshot();
    });
  });
  describe('removeInvalidExclusionsFromFb', () => {
    it('should remove invalid segments from exclusions', () => {
      const mockAudience = cloneDeep(mockSearchResults[1]);
      (SearchUtils as any).removeInvalidExclusionsFromFb(mockAudience, ['6003109292833'], ['demographics']);
      expect(mockAudience).toMatchSnapshot();
    });
  });
});
