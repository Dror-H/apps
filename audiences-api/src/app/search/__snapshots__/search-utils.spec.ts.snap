// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search-utils removeEmptyObjectsAndArraysFromTargeting should remove empty objects and arrays from flexible specs and exclusions 1`] = `
Object {
  "createdByContributors": false,
  "id": "478605de-ba43-5d8c-9c52-b1043e4cc0b7",
  "segments": Array [
    "SO - Conversions - Bracelet - CBO - UK Only",
    "Interest Jewellry - UK - 18+ – Copy",
    "Jewellery",
  ],
  "spec": Object {
    "campaign": Object {
      "id": "23848984859400054",
      "name": "SO - Conversions - Bracelet - CBO - UK Only",
    },
    "id": "23848984859650054",
    "name": "Interest Jewellry - UK - 18+ – Copy",
    "targeting": Object {
      "age_max": 65,
      "age_min": 35,
      "brand_safety_content_filter_levels": Array [
        "FACEBOOK_STANDARD",
        "AN_STANDARD",
      ],
      "exclusions": undefined,
      "flexible_spec": Array [],
      "genders": Array [
        2,
      ],
      "geo_locations": Object {
        "countries": Array [
          "GB",
        ],
        "location_types": Array [
          "home",
          "recent",
        ],
      },
    },
  },
  "topics": Array [
    "uk",
    "jewellry",
    "jewellery",
    "interest",
    "copy",
  ],
  "userTags": Array [],
}
`;

exports[`search-utils removeInvalidExclusionsFromFb should remove invalid segments from exclusions 1`] = `
Object {
  "createdByContributors": false,
  "id": "478605de-ba43-5d8c-9c52-b1043e4cc0b7",
  "segments": Array [
    "SO - Conversions - Bracelet - CBO - UK Only",
    "Interest Jewellry - UK - 18+ – Copy",
    "Jewellery",
  ],
  "spec": Object {
    "campaign": Object {
      "id": "23848984859400054",
      "name": "SO - Conversions - Bracelet - CBO - UK Only",
    },
    "id": "23848984859650054",
    "name": "Interest Jewellry - UK - 18+ – Copy",
    "targeting": Object {
      "age_max": 65,
      "age_min": 35,
      "brand_safety_content_filter_levels": Array [
        "FACEBOOK_STANDARD",
        "AN_STANDARD",
      ],
      "exclusions": Object {
        "interests": Array [
          Object {
            "id": "6003109292833",
            "name": "Drop shipping",
          },
        ],
      },
      "flexible_spec": Array [
        Object {
          "interests": Array [
            Object {
              "id": "6003266225248",
              "name": "Jewellery",
            },
          ],
        },
      ],
      "genders": Array [
        2,
      ],
      "geo_locations": Object {
        "countries": Array [
          "GB",
        ],
        "location_types": Array [
          "home",
          "recent",
        ],
      },
    },
  },
  "topics": Array [
    "uk",
    "jewellry",
    "jewellery",
    "interest",
    "copy",
  ],
  "userTags": Array [],
}
`;

exports[`search-utils removeInvalidFlexibleSpecFromFb should remove invalid segments from flexible spec 1`] = `
Object {
  "createdByContributors": false,
  "id": "4870eab0-7ed2-56b1-9493-5b4684a7b03c",
  "segments": Array [],
  "spec": Object {
    "id": "0478f7d5-55c7-4106-ba34-4f691236969c",
    "name": "",
    "targeting": Object {
      "age_max": 65,
      "age_min": 18,
      "excluded_custom_audiences": Array [],
      "excluded_geo_locations": Object {},
      "exclusions": Object {},
      "flexible_spec": Array [
        Object {
          "work_positions": Array [
            Object {
              "id": "105703889464434",
              "name": "Motivational speaker",
            },
          ],
        },
      ],
      "genders": Array [
        0,
      ],
      "geo_locations": Object {
        "countries": Array [
          "US",
        ],
        "location_types": Array [
          "home",
          "recent",
        ],
      },
      "locales": Array [],
      "targeting_optimization": "none",
    },
  },
  "topics": Array [],
  "userTags": Array [],
}
`;
